src/coreclr/vm/arm64/thunktemplates.S

```diff
@@ -62,10 +62,10 @@ LEAF_END_MARKED StubPrecodeCodeTemplate, _TEXT
 
     ldr x11, DATA_SLOT(FixupPrecode, Target, FIXUP_PRECODE_CODESIZE, FixupPrecodeCodeTemplate)
     br  x11
+    dmb ishld
```

Comment:
Should this be removed now that we use the cache flushing?

Reply:
No, that's still necessary. We're not flushing around creation of `FixupPrecode`, since it's expensive and I was able to come up with an alternative approach for that.

Reply:
> we can slide the barrier into only the slow path as long as we ensure that the Target field written with a VolatileStore after the MethodDesc and PrecodeFixupThunk fields are set

This guarantees ordering between Target and MethodDesc+PrecodeFixupThunk fields. What guarantees that the executing code is going to see Target in this scheme?

Reply:
Before we map the memory for the executable section of the FixupPrecodeThunks into the process, we fill the Target with pointers so that Target points at the second portion of the FixupPrecodeThunk. So when we map the executable memory in, we get a natural FlushInstructionCache for that scenario. So there is never a case where we need to worry about the Target field being set to NULL. Then we either are able to see the actual final Target (which doesn't require any further synchronization), or we'll hit the memory barrier in the second portion of the FixupPrecodeThunk and find that the MethodDesc/PrecodeFixupThunk are properly set. This deserves a better comment if you couldn't find that though.